wled00/wled.cpp

```diff
@@ -915,6 +917,7 @@ void WLED::handleConnection()
     if (now - lastReconnectAttempt > ((stac) ? 300000 : 18000) && WLED_WIFI_CONFIGURED) {
       if (improvActive == 2) improvActive = 3;
       DEBUG_PRINTLN(F("Last reconnect too old."));
+      if (++selectedWiFi >= multiWiFi.size()) selectedWiFi = 0;
```

Comment:
Networks are retried every 18s in current implementation (after unsuccessful attempt) with 3 possible configured networks a connection will be established within a minute. It may not be the fastest but it will still work as intended and there are no consequences.

The problem with your code is that it tries to duplicate existing connectivity approach. That should not be done as there need to be exactly one location a connection is established as many things rely on that (including notification to usermods).

If you really do not want to wait for a minute on your third network, you could adapt the scan to proactively select appropriate network (`selectedWiFi`) and submit a PR to this branch (co-authoring this PR). I am also not sure if scanning for available networks constantly is really the best approach as it must take away a few CPU cycles.

Reply:
You're trying to connect to a list of network following its index instead of which is nearest, which results not only the third one not being connected immediately, but also connecting to the worst network in most cases.

Let's think that I have a bracelet (which I'm making and that's why I made my PR) and I have saved both my wifi and my neighbor's. As of course my WiFi will be the first on the list, every time I turn on my bracelet, it'll try to connect to my WiFi, and then to my neighbor's if it fails. The issue comes when I go to my neighbor house, because my WiFi will still be visible and connectable, but with almost no signal, making almost not reachable by my phone.

Making things worse, by normal behavior, the ESP will not recognize it lost WiFi connection after a while, maybe not an entire minute, but here every seconds matter with your approach.

And, making this even worse, as `WLED_MAX_WIFI_COUNT` can be configurable, I can set it to 5, or to 10 if I have a board that support it, so now it's 18 * 10 = 180 seconds = 3 minutes to connect to the last WiFi network, which is unacceptable by any mean.

If you don't want to loose CPU cycles with `Scan()`, then make a WiFi list that it's being updated every time `selectedWiFi` reach 0, with that, we can assure to connect not only to a reachable network, but also to the nearest.

Reply:
That's *unacceptable* for your particular case, not any case. But I hear you.

Feel free to optimise.

Reply:
Latest commit adds prioritisation of WiFi-s according to signal strength and order of importance.